contracts/src/spaces/facets/roles/RolesBase.sol

```diff
@@ -318,6 +324,91 @@ abstract contract RolesBase is IRolesBase {
     emit RoleRemoved(msg.sender, roleId);
   }
 
+  // =============================================================
+  // Channel Permissions
+  // =============================================================
+  function _getChannelPermissionOverrides(
+    uint256 roleId,
+    bytes32 channelId
+  ) internal view returns (string[] memory permissions) {
+    // check role exists
+    _checkRoleExists(roleId);
+
+    // check channel exists
+    ChannelService.checkChannelExists(channelId);
+
+    return
+      RolesStorage
+      .layout()
+      .permissionByChannelIdByRoleId[roleId][channelId].values();
+  }
+
+  function _setChannelPermissionOverrides(
+    uint256 roleId,
+    bytes32 channelId,
+    string[] memory permissions
+  ) internal {
+    ChannelService.checkChannelExists(channelId);
+
+    // check role exists
+    _checkRoleExists(roleId);
+
+    RolesStorage.Layout storage rs = RolesStorage.layout();
+
+    rs.channelsByRole[roleId].add(channelId);
+
+    StringSet.Set storage permissionsSet = rs.permissionByChannelIdByRoleId[
+      roleId
+    ][channelId];
+
+    // remove current channel permissions if any
+    if (permissionsSet.length() > 0) {
+      string[] memory currentPermissions = permissionsSet.values();
+      uint256 currentPermissionsLen = currentPermissions.length;
+      for (uint256 i = 0; i < currentPermissionsLen; i++) {
+        permissionsSet.remove(currentPermissions[i]);
+      }
+    }
+
+    // check if new permissions are not empty then add them
+    uint256 permissionsLen = permissions.length;
+    for (uint256 i = 0; i < permissionsLen; i++) {
+      _checkEmptyString(permissions[i]);
+      if (!permissionsSet.add(permissions[i])) {
+        revert Roles__PermissionAlreadyExists();
+      }
+    }
+
+    emit PermissionsAddedToChannelRole(msg.sender, roleId, channelId);
+  }
+
+  function _clearChannelPermissionOverrides(
+    uint256 roleId,
+    bytes32 channelId
+  ) internal {
+    // check role exists
+    _checkRoleExists(roleId);
+
+    // check channel exists
+    ChannelService.checkChannelExists(channelId);
+
+    RolesStorage.Layout storage rs = RolesStorage.layout();
+    StringSet.Set storage permissionsSet = rs.permissionByChannelIdByRoleId[
+      roleId
+    ][channelId];
+
+    // get current permissions
+    string[] memory currentPermissions = permissionsSet.values();
+    uint256 currentPermissionsLen = currentPermissions.length;
+    for (uint256 i = 0; i < currentPermissionsLen; i++) {
+      permissionsSet.remove(currentPermissions[i]);
+    }
+
+    rs.channelsByRole[roleId].remove(channelId);
```

Comment:
So, question: if a role applies to a channel and we remove the override permissions, but the role still applies to the channel with it's default permissions, we would not want to remove it from this map here, right? 

I think I didn't clearly get at this, but I'm wondering if adding or removing channels from this map actually makes sense at all from these entrypoints...

Reply:
this is strictly for override so we can check if there exists some settings that were placed there, entitlement data queryable uses this storage check to diff between global and channel level permissions